packages/axios-extension/src/abap/adt-catalog/services/transportrequest-service.ts

```diff
@@ -26,24 +23,34 @@ export class TransportRequestService extends AdtService {
     }
 
     /**
+     * TransportRequestService API function to create a new transport number.
      *
+     * @param packageName Package name for deployment
+     * @param appName Fiori project name for deployment. A new project that has not been deployed before is also allowed
      * @param description Description of the new transport request to be created
      * @returns Newly created transport request number
      */
-    public async createTransportRequest(description: string): Promise<string> {
+    public async createTransportRequest(packageName: string, appName: string, description: string): Promise<string> {
```

Comment:
@slin-sap, please consider using a destructured object for the parameters here - the number is starting to get unwieldy.

Also, can the parameter `appName` be called `ui5AppName` or something for clarity?

Reply:
Updated param name

Reply:
Thank you for renaming that.

Can you do the destructured object for the parameters too? Positional arguments are a hidden and subtle form of tight coupling - the order of the parameters being the main cause. It's easy to transpose two parameters of the same type too, for example. The problem gets worse with increasing number of arguments. If you look at the other code in `axios-extension`, most of the methods take 0 or 1 arguments.

Typescript does not have named parameters like Python does and the using destructured parameters is a good alternative.

This is available as an automated refactoring in VS Code (it's also the only automated refactoring offered. Food for thought?). VS Code takes care of changing the code in the invocation locations as well.